frontend/depgraph.lua

```diff
@@ -45,72 +93,92 @@ function DepGraph:removeNode(node_key)
     -- one in their override=, that have added themselves in
     -- this node's .deps). We don't want to lose these
     -- dependencies if we later re-addNode this node.
-    local node = self.nodes[node_key]
+    local node, index = self:getNode(node_key)
     if node then
         if not node.deps or #node.deps == 0 then
-            self.nodes[node_key] = nil
+            -- We need to clear *all* references to that node for it to be GC'ed...
+            node = nil --luacheck: ignore
```

Comment:
Why do you insist on `node = nil` (even luacheck tells you it's not needed)?  It's a local, it will get nil when out of scope.
(We usually have to do it for self.something = nil to be gc'ed, but there, I don't see why.)

Reply:
Because it irks me that this one annoys luacheck, but not the two others below ;p.

It is indeed completely superfluous, as are the other two, especially since those local's are never re-used later in the code.

(i.e., it's more a pedantic point than anything)

Reply:
Okay turns out it's actively harmful because it does make the array sparse ;p.

Reply:
And as `ipairs` stops at the first hole (i.e., `nil` or hash), thats Bad(TM).

Reply:
And `table.remove` is also useless with sparse arrays (i.e., can't drop an index set to nil).

Reply:
Not sure I knew about all that :) but yes, generally, the simpler the better :)